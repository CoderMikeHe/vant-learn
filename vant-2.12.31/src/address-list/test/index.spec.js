import { mount } from '../../../test';
import AddressList from '..';

const list = [
  {
    id: '1',
    name: '张三',
    tel: '13000000000',
    address: '浙江省杭州市西湖区文三路 138 号东方通信大厦 7 楼 501 室',
  },
  {
    id: '2',
    name: '李四',
    tel: '1310000000',
    address: '浙江省杭州市拱墅区莫干山路 50 号',
  },
];

test('unswitchable', () => {
  const wrapper = mount(AddressList, {
    propsData: {
      list,
      switchable: false,
    },
  });

  expect(wrapper).toMatchSnapshot();
});

test('select event', () => {
  const onSelect = jest.fn();
  const wrapper = mount(AddressList, {
    propsData: {
      list,
    },
    context: {
      on: {
        select: onSelect,
      },
    },
  });

  wrapper.find('.van-radio__icon').trigger('click');

  expect(onSelect).toHaveBeenCalledTimes(1);
});

test('click-item event', () => {
  const onClickItem = jest.fn();
  const wrapper = mount(AddressList, {
    propsData: {
      list,
    },
    context: {
      on: {
        'click-item': onClickItem,
      },
    },
  });

  wrapper.find('.van-address-item').trigger('click');

  expect(onClickItem).toHaveBeenCalledTimes(1);
});

test('should render tag  slot correctly', () => {
  const wrapper = mount(AddressList, {
    scopedSlots: {
      tag: () => 'Custom Tag',
    },
  });
  expect(wrapper.html()).toMatchSnapshot();
});
