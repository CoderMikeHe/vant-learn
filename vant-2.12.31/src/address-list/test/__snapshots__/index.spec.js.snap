// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render tag  slot correctly 1`] = `
<div class="van-address-list">
  <div role="radiogroup" class="van-radio-group"></div>
  <div class="van-address-list__bottom"><button class="van-button van-button--danger van-button--normal van-button--block van-button--round van-address-list__add">
      <div class="van-button__content"><span class="van-button__text">新增地址</span></div>
    </button></div>
</div>
`;

exports[`unswitchable 1`] = `
<div class="van-address-list">
  <div role="radiogroup" class="van-radio-group">
    <div class="van-address-item">
      <div class="van-cell van-cell--borderless">
        <div class="van-cell__value van-cell__value--alone van-address-item__value">
          <div class="van-address-item__name">张三 13000000000</div>
          <div class="van-address-item__address">浙江省杭州市西湖区文三路 138 号东方通信大厦 7 楼 501 室</div>
        </div><i class="van-icon van-icon-edit van-address-item__edit">
          <!----></i>
      </div>
    </div>
    <div class="van-address-item">
      <div class="van-cell van-cell--borderless">
        <div class="van-cell__value van-cell__value--alone van-address-item__value">
          <div class="van-address-item__name">李四 1310000000</div>
          <div class="van-address-item__address">浙江省杭州市拱墅区莫干山路 50 号</div>
        </div><i class="van-icon van-icon-edit van-address-item__edit">
          <!----></i>
      </div>
    </div>
  </div>
  <div class="van-address-list__bottom"><button class="van-button van-button--danger van-button--normal van-button--block van-button--round van-address-list__add">
      <div class="van-button__content"><span class="van-button__text">新增地址</span></div>
    </button></div>
</div>
`;
